### ISSUE TITLE
Certain biomes cause a crash

### ISSUE THREAD

**Issue by FinishedFragment** [minetest-LOTR/Lord-of-the-Test#167]
**Steps to reproduce:**

- Create a new world
- Pick the mordor orc as your race
- The subgame crashes because of mapgen errors. Log attached

[crash.log](https://github.com/minetest-LOTR/Lord-of-the-Test/files/5674345/crash.log)

**Comment by FinishedFragment** [comment]
Seems like some of the biomes are buggy?
Three more crashes:

```
ERROR[Main]: ServerError: AsyncErr: Lua: finishGenRuntime error from mod 'lottmapgen' in callback environment_OnGenerated(): Unknown node: lottplants:pipeweed_wild
ERROR[Main]: stack traceback:
ERROR[Main]: 	[C]: in function 'get_content_id'
ERROR[Main]: 	...Lord-of-the-Test/mods/lottmapgen/technical/functions.lua:46: in function 'farming_plants'
ERROR[Main]: 	...of-the-Test/mods/lottmapgen/mapgen/biomes/minhiriath.lua:91: in function 'deco'
ERROR[Main]: 	...games/Lord-of-the-Test/mods/lottmapgen/mapgen/mapgen.lua:239: in function <...games/Lord-of-the-Test/mods/lottmapgen/mapgen/mapgen.lua:101>
ERROR[Main]: 	/usr/share/minetest/builtin/game/register.lua:429: in function </usr/share/minetest/builtin/game/register.lua:413>
```

```
ERROR[Main]: ServerError: AsyncErr: Lua: finishGenRuntime error from mod 'lottmapgen' in callback environment_OnGenerated(): Unknown node: lottplants:young_mallorn_leaves
ERROR[Main]: stack traceback:
ERROR[Main]: 	[C]: in function 'get_content_id'
ERROR[Main]: 	...mes/Lord-of-the-Test/mods/lottmapgen/technical/pools.lua:47: in function 'remove_tree'
ERROR[Main]: 	...mes/Lord-of-the-Test/mods/lottmapgen/technical/pools.lua:195: in function 'generate_pools'
ERROR[Main]: 	...games/Lord-of-the-Test/mods/lottmapgen/mapgen/mapgen.lua:290: in function <...games/Lord-of-the-Test/mods/lottmapgen/mapgen/mapgen.lua:101>
ERROR[Main]: 	/usr/share/minetest/builtin/game/register.lua:429: in function </usr/share/minetest/builtin/game/register.lua:413>
```

```
ERROR[Main]: ServerError: AsyncErr: Lua: finishGenRuntime error from mod 'lottmapgen' in callback environment_OnGenerated(): Unknown node: lottplants:apple
ERROR[Main]: stack traceback:
ERROR[Main]: 	[C]: in function 'get_content_id'
ERROR[Main]: 	...Lord-of-the-Test/mods/lottmapgen/technical/functions.lua:333: in function 'generate_tree'
ERROR[Main]: 	...d-of-the-Test/mods/lottmapgen/mapgen/biomes/breeland.lua:25: in function 'deco'
ERROR[Main]: 	...games/Lord-of-the-Test/mods/lottmapgen/mapgen/mapgen.lua:239: in function <...games/Lord-of-the-Test/mods/lottmapgen/mapgen/mapgen.lua:101>
ERROR[Main]: 	/usr/share/minetest/builtin/game/register.lua:429: in function </usr/share/minetest/builtin/game/register.lua:413>
```

It seems like the start biomes of the following races cause crashes:

- Bree Human/Hobbit
- Mordor Orc
- Rhun Human
- Dunland Human
- Lorien Elf

**Steps to reproduce:**

I just chose a race and walked around in the biome for a few minutes.

**Comment by lumidify** [comment]
Hi,
As I wrote on your other issue, this branch is very experimental, so many things will crash. A lot of code here was copied from the original LOTT but some required items haven't been copied yet. As far as I know, Minetest used to be a lot more lenient about that, which is probably why it wasn't noticed, but now it crashes when something like that happens (which is how it should be). If you want something that at least somewhat works, use the old version of LOTT since (sadly) no one will have time to fix any of this properly in the near future. If you want to, you're more than welcome to fix it yourself and submit a pull request, though.

**Comment by FinishedFragment** [comment]
I just wanted to get a "sneak peak" at the rewrite and its looking pretty decent (except the things I have reported). Awesome work so far, its very detailed.
Unfortunately I do not have much lua or minetest modding experience but I found several typos in the race selection menu. These should be pretty easy to fix as a lua noob, so I might just go ahead and submit a PR soon™ to fix these typos.
